### PR TITLE
Expose `randfn` to global scope

### PR DESCRIPTION
**Pull request by Chaosus**
I think it's not that bad to make it globally accessed as I did for `randi_range` previously simply for the user's comfortability.

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs [hunk 1]
```diff
@@ -238,6 +243,9 @@ public static Variant.Type TypeToVariantType(Type type)
         [MethodImpl(MethodImplOptions.InternalCall)]
         internal extern static double godot_icall_GD_randf_range(double from, double to);
 
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal extern static float godot_icall_GD_randfn();
```
Comment (CedNaru) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs:247]: I know nothing about C# interop but aren't you forgetting the parameters here?

Comment (Chaosus) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs]: Yeah, thanks

File: modules/visual_script/doc_classes/VisualScriptBuiltinFunc.xml [hunk 2]
```diff
@@ -120,102 +120,105 @@
 		<constant name="MATH_RANDF_RANGE" value="34" enum="BuiltinFunc">
 			Return a random floating-point value between the two inputs.
 		</constant>
-		<constant name="MATH_RANDI_RANGE" value="35" enum="BuiltinFunc">
+		<constant name="MATH_RANDFN" value="35" enum="BuiltinFunc">
+			Returns a normally-distributed preudo-random floating point value using Box-Muller transform with the specified [code]mean[/code] and a standard [code]deviation[/code]. This is also called Gaussian distribution.
```
Comment (aaronfranke) [inline modules/visual_script/doc_classes/VisualScriptBuiltinFunc.xml:124]: Typo, "preudo".

Comment (Chaosus) [reply modules/visual_script/doc_classes/VisualScriptBuiltinFunc.xml]: Fixed

**Comment by Xrayez** [comment]
It makes sense to me to provide most `RandomPCG` functionality to global scope for completeness sake, especially if we don't plan to provide a global instance of `RandomNumberGenerator`: https://github.com/godotengine/godot-proposals/issues/1741#issuecomment-717944646.

File: modules/visual_script/visual_script_builtin_funcs.h [hunk 3]
```diff
@@ -72,6 +72,7 @@ class VisualScriptBuiltinFunc : public VisualScriptNode {
 		MATH_RANDI,
 		MATH_RANDF,
 		MATH_RANDF_RANGE,
+		MATH_RANDFN,
 		MATH_RANDI_RANGE,
```
Comment (akien-mga) [inline modules/visual_script/visual_script_builtin_funcs.h:76]: The order is a bit awkward, I would change it to match `variant_utility`:
```suggestion
		MATH_RANDI,
		MATH_RANDF,
		MATH_RANDFN,
		MATH_RANDI_RANGE,
		MATH_RANDF_RANGE,
```
And thus moving stuff around elsewhere to match that order.

Comment (akien-mga) [reply modules/visual_script/visual_script_builtin_funcs.h:76]: Note that my suggestion also swaps `RANDI_RANGE` and `RANDF_RANGE` which were not in the same order as `RANDI` and `RANDF`.

Could also be this everywhere:
```
randi
randf
randi_range
randf_range
randfn
```

Comment (Chaosus) [reply modules/visual_script/visual_script_builtin_funcs.h]: Done (I think)

File: modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs [hunk 4]
```diff
@@ -322,6 +322,16 @@ public static float Randf()
             return godot_icall_GD_randf();
         }
 
+        /// <summary>
+        /// Returns a normally-distributed pseudo-random number, using Box-Muller transform with the specified <c>mean</c> and a standard <c>deviation</c>.
+        /// This is also called Gaussian distribution.
+        /// </summary>
+        /// <returns>A random normally-distributed <see langword="float"/> number.</returns>
+        public static double Randfn(double mean, double deviation)
```
Comment (akien-mga) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs:330]: I'd move it after `RandRange(int, int)` for consistency.

The order is a bit different in C# it seems as it does:
```
randf
randi
randf_range
randi_range
```
but that's probably not a big deal :)

File: modules/mono/glue/gd_glue.cpp [hunk 5]
```diff
@@ -198,6 +198,10 @@ double godot_icall_GD_randf_range(double from, double to) {
 	return Math::random(from, to);
 }
 
+double godot_icall_GD_randfn(double mean, double deviation) {
```
Comment (akien-mga) [inline modules/mono/glue/gd_glue.cpp:201]: Move it after `randi_range`.

File: modules/visual_script/visual_script_builtin_funcs.h [hunk 6]
```diff
@@ -71,8 +71,9 @@ class VisualScriptBuiltinFunc : public VisualScriptNode {
 		MATH_RANDOMIZE,
 		MATH_RANDI,
 		MATH_RANDF,
-		MATH_RANDF_RANGE,
+		MATH_RANDFN,
```
Comment (akien-mga) [inline modules/visual_script/visual_script_builtin_funcs.h:74]: To match the other changes you've done it should go after `MATH_RANDF_RANGE` now.

Comment (akien-mga) [reply modules/visual_script/visual_script_builtin_funcs.h]: Fixed already.

**Review by akien-mga — APPROVED** [review]

File: modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs (hunk 4 above)
Comment (Chaosus) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs:330]: I think it's sorted alphabetically in the C# file rather than that.

Comment (akien-mga) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs:330]: Good catch.

**Comment by akien-mga** [comment]
Mono is failing:
```
/home/runner/work/godot/godot/modules/mono/glue/GodotSharp/GodotSharp/Core/GD.cs(332,20): error CS0103: The name 'godot_icall_GD_randfn' does not exist in the current context [/home/runner/work/godot/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
```

**Review by mhilbrunner — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**